library/Rattletrap/Console/Main.hs

```diff
@@ -32,22 +35,34 @@ rattletrap name arguments = do
   Monad.when (configHelp config) (printHelp name *> Exit.exitFailure)
   Monad.when (configVersion config) (printVersion *> Exit.exitFailure)
   input <- getInput config
-  let decode = getDecoder config
-  replay <- either fail pure (decode input)
-  let encode = getEncoder config
-  putOutput config (encode replay)
-
-getDecoder :: Config -> Bytes.ByteString -> Either String Rattletrap.Replay
-getDecoder config = case getMode config of
-  ModeDecode -> Rattletrap.decodeReplayFile
-  ModeEncode -> Rattletrap.decodeReplayJson
-
-getEncoder :: Config -> Rattletrap.Replay -> Bytes.ByteString
-getEncoder config = case getMode config of
-  ModeDecode -> if configCompact config
-    then LazyBytes.toStrict . Json.encode
-    else Rattletrap.encodeReplayJson
-  ModeEncode -> Rattletrap.encodeReplayFile
+  let outputResult = getTransformation config input
+  output <- either fail return outputResult
+  putOutput config output
+
+getTransformation :: Config -> Bytes.ByteString -> Either String Bytes.ByteString
+getTransformation config =
+  if configHeaderOnly config
+  then
+    let (decoder, encoder) = case getMode config of
+            ModeDecode ->
+              let e = if configCompact config
+                      then LazyBytes.toStrict . Json.encode
+                      else Rattletrap.encodeReplayJson
+              in (Common.runDecode (decodeSection Header.decodeHeader), e)
+            ModeEncode -> ( const $ Left "Unable to use header-only mode when encoding replay file"
```

Comment:
It should be possible to generate a header-only replay by setting all of the `Content` fields to some empty value.